### DISCUSSION TITLE
Multiple Instances or Server connection function (Cluster)

### DISCUSSION THREAD

**Discussion by Tizian-Maxime-Weigt** [NginxProxyManager/nginx-proxy-manager#3152] [Ideas]
it would be a cool idea if you can build a cluster or have a way to connect multiple servers

**Comment by olariuromeo** [comment]
I think it is possible if you add a load balancer in front I think it is possible if you add a load balancer in front, which distributes the traffic on each instance

**Reply by Tizian-Maxime-Weigt** [reply]
yes, you're right, but I wanted to build something like Cloudflare with Nginx. I've finally written a complete system of my own in Python with ddos protection on a layer7 basis, internal and external loabbalencer and master and worker system, etc.

**Comment by mike548141** [comment]
I would be more interested in NPM being a load balancer for the hosts it sits in front of. This would add a major feature to NPM, and partially fulfill what you are looking for e.g. you could have a load balancing NPM in front of two other NPM's, obviously still with a single point of failure. The main difference between this and what you are asking would be session synchronisation between the two NPM hosts.

@Tizian-Maxime-Weigt A thought, have you considered using round robin DNS? If you have two public IP addresses, route one IP to each NPM host. Then in DNS configure two A records, both with the same name but a different public IP. e.g.
www.example.com --> 1.2.3.4
www.example.com --> 5.6.7.8
When someone looks up www.example.com they will receive both IP's and can connect to either. Then its a matter of de-registering a host if it goes offline.

**Comment by Tizian-Maxime-Weigt** [comment]
Hi there,

I was thinking of a self-hosted solution from Cloudflare and I commissioned a developer to do it myself...

Kind regards,
Titian Maxime Weigt

**Comment by vincent1890** [comment]
**Multiple instances synchronize or a cluster function** 
As the example of project Adguard https://github.com/bakito/adguardhome-sync which allows to synchronize two hote Adguard would be really grandiose for Nginx Proxy Manager

**Comment by vincent1890** [comment]
> it would be a cool idea if you can build a cluster or have a way to connect multiple servers

Multiple instances synchronize or a cluster function 
As the example of project Adguard https://github.com/bakito/adguardhome-sync which allows to synchronize two hote Adguard would be really grandiose for Nginx Proxy Manager
